cnf-certification-test/lifecycle/podsets/podsets.go

```diff
@@ -0,0 +1,147 @@
+	notReady := true
+	for _, condition := range deployment.Status.Conditions {
+		if condition.Type == v1app.DeploymentAvailable {
+			notReady = false
+			break
+		}
+	}
+	var replicas int32
+	if deployment.Spec.Replicas != nil {
+		replicas = *(deployment.Spec.Replicas)
+	} else {
+		replicas = 1
+	}
+	if notReady ||
+		deployment.Status.UnavailableReplicas != 0 ||
+		deployment.Status.ReadyReplicas != replicas ||
+		deployment.Status.AvailableReplicas != replicas ||
+		deployment.Status.UpdatedReplicas != replicas {
+		return false
+	}
+	return true
+}
+
+func WaitForStatefulSetReady(ns, name string, timeout time.Duration) bool { //nolint:dupl // not duplicate
+	logrus.Trace("check if statefulset ", ns, ":", name, " is ready ")
+	clients := clientsholder.GetClientsHolder()
+	start := time.Now()
+	for time.Since(start) < timeout {
+		ss, err := provider.GetUpdatedStatefulset(clients.AppsClients, ns, name)
+		if err == nil && IsStatefulSetReady(ss) {
+			logrus.Trace("statefulset ", ns, ":", name, " is ready ")
+			return true
+		}
+		time.Sleep(time.Second)
+	}
+	logrus.Error("statefulset ", ns, ":", name, " is not ready ")
+	return false
+}
+
+func IsStatefulSetReady(statefulset *v1app.StatefulSet) bool {
+	var replicas int32
+	if statefulset.Spec.Replicas != nil {
+		replicas = *(statefulset.Spec.Replicas)
+	} else {
+		replicas = 1
+	}
+	if statefulset.Status.ReadyReplicas != replicas ||
+		statefulset.Status.CurrentReplicas != replicas ||
+		statefulset.Status.UpdatedReplicas != replicas {
+		return false
+	}
+	return true
+}
+
+func WaitForAllPodSetReady(env *provider.TestEnvironment, timeoutPodSetReady time.Duration) (claimsLog loghelper.CuratedLogLines) {
+	for _, dut := range env.Deployments {
+		isReady := WaitForDeploymentSetReady(dut.Namespace, dut.Name, timeoutPodSetReady)
+		if isReady {
+			claimsLog = claimsLog.AddLogLine("deployment: %s Status: OK", provider.DeploymentToString(dut))
+		} else {
+			claimsLog = claimsLog.AddLogLine("deployment: %s Status: NOK", provider.DeploymentToString(dut))
+		}
+	}
+	for _, sut := range env.SatetfulSets {
+		isReady := WaitForDeploymentSetReady(sut.Namespace, sut.Name, timeoutPodSetReady)
+		if isReady {
+			claimsLog = claimsLog.AddLogLine("statefulset: %s Status: OK", provider.StatefulsetToString(sut))
+		} else {
+			claimsLog = claimsLog.AddLogLine("statefulset: %s Status: NOK", provider.StatefulsetToString(sut))
+		}
+	}
+	return claimsLog
+}
+
+func GetPodSetNodes(pods []*v1.Pod, ssName string, nodesIn map[string]bool) map[string]bool {
+	for _, put := range pods {
+		for _, or := range put.OwnerReferences {
+			if or.Kind != ReplicaSetString && or.Kind != StatefulsetString {
+				continue
+			}
+			nodesIn[put.Spec.NodeName] = true
+			break
+		}
+	}
+	return nodesIn
+}
+
+func GetAllNodesForAllPodSets(env *provider.TestEnvironment) (nodes map[string]bool) {
+	nodes = make(map[string]bool)
+	for _, dut := range env.Deployments {
+		nodes = GetPodSetNodes(env.Pods, dut.Name, nodes)
```

Comment:
Only one call to GetPodSetNodes is needed here. dut/sut names are not needed either.

Reply:
fixed